api/src/DataPersister/InvitationDataPersister.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\DataPersister;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use App\DTO\Invitation;
+use App\Entity\CampCollaboration;
+use App\Repository\CampCollaborationRepository;
+use App\Repository\UserRepository;
+use App\Util\ValidationExceptionFactory;
+use Symfony\Component\Security\Core\Security;
+
+class InvitationDataPersister implements ContextAwareDataPersisterInterface {
+    public function __construct(
+        private ContextAwareDataPersisterInterface $dataPersister,
+        private CampCollaborationRepository $campCollaborationRepository,
+        private Security $security,
+        private UserRepository $userRepository
+    ) {
+    }
+
+    public function supports($data, array $context = []): bool {
+        return $data instanceof Invitation;
+    }
+
+    /**
+     * @param Invitation $data
+     *
+     * @throws ResourceClassNotSupportedException
+     *
+     * @return object|void
+     */
+    public function persist($data, array $context = []) {
+        if (Invitation::ACCEPT === ($context['item_operation_name'] ?? null)) {
+            if ($data->userAlreadyInCamp) {
+                $message = "The user {$data->userDisplayName} is already associated with the camp {$data->campTitle}";
+
+                throw ValidationExceptionFactory::createValidationException($message);
+            }
+
+            $campCollaboration = $this->campCollaborationRepository->findByInviteKey($data->inviteKey);
+            $campCollaboration->user = $this->security->getUser();
+            $campCollaboration->status = CampCollaboration::STATUS_ESTABLISHED;
+            $campCollaboration->inviteKey = null;
+            $campCollaboration->inviteEmail = null;
+            //TODO: add MaterialList for User
+            $this->dataPersister->persist($campCollaboration);
+
+            $data->userAlreadyInCamp = true;
```

Comment:
Nötig?

Reply:
Ist so oder so die Frage, weil nach accept oder reject findet man die Invitation nicht mehr... das ist bisschen komisch.
Ich hätte es jetzt so gemacht, weil es den Stand eher abbildet (Der Benutzer ist jetzt im camp).

Reply:
Aber wird denn das persistiert? Ich hatte den eindruck dass danach nie mehr etwas mit $data passiert

Reply:
Nein, Invitation wird nie persistiert, es ist ein DTO das aus dem eingeloggten User und der im Request referenzierten CampCollaboration zusammengebaut wird.
Es wird aufgrund von Operations die CampCollaboration geändert, aber nie direkt eine Invitation (mit mapping) in die Datenbank geschrieben.
$data wird deserialisiert in der Response zurückgegeben. (Siehe https://github.com/ecamp/ecamp3/pull/1859/files#diff-74c41eaa1f7f42a90ba91740dcde7f1ca280e0d6960a0061070c863e0bf4a4beR58).